spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,345 @@
+          } else {
+            return null;
+          }
+
+        case GT:
+          if (isRef(leftChild(predicate)) && isLiteral(rightChild(predicate))) {
+            String columnName = toColumnName(leftChild(predicate));
+            return greaterThan(columnName, convertLiteral(rightChild(predicate)));
+          } else if (isRef(rightChild(predicate)) && isLiteral(leftChild(predicate))) {
+            String columnName = toColumnName(rightChild(predicate));
+            return lessThan(columnName, convertLiteral(leftChild(predicate)));
+          } else {
+            return null;
+          }
+
+        case GT_EQ:
+          if (isRef(leftChild(predicate)) && isLiteral(rightChild(predicate))) {
+            String columnName = toColumnName(leftChild(predicate));
+            return greaterThanOrEqual(columnName, convertLiteral(rightChild(predicate)));
+          } else if (isRef(rightChild(predicate)) && isLiteral(leftChild(predicate))) {
+            String columnName = toColumnName(rightChild(predicate));
+            return lessThanOrEqual(columnName, convertLiteral(leftChild(predicate)));
+          } else {
+            return null;
+          }
+
+        case EQ: // used for both eq and null-safe-eq
+          Object value;
+          String columnName;
+          if (isRef(leftChild(predicate)) && isLiteral(rightChild(predicate))) {
+            columnName = toColumnName(leftChild(predicate));
+            value = convertLiteral(rightChild(predicate));
+          } else if (isRef(rightChild(predicate)) && isLiteral(leftChild(predicate))) {
+            columnName = toColumnName(rightChild(predicate));
+            value = convertLiteral(leftChild(predicate));
+          } else {
+            return null;
+          }
+
+          if (predicate.name().equals(EQ)) {
+            // comparison with null in normal equality is always null. this is probably a mistake.
+            Preconditions.checkNotNull(
+                value, "Expression is always false (eq is not null-safe): %s", predicate);
+            return handleEqual(columnName, value);
+          } else { // "<=>"
+            if (value == null) {
+              return isNull(columnName);
+            } else {
+              return handleEqual(columnName, value);
+            }
+          }
+
+        case IN:
+          if (isSupportedInPredicate(predicate)) {
+            return in(
+                toColumnName(childAtIndex(predicate, 0)),
+                Arrays.stream(predicate.children())
+                    .skip(1)
+                    .map(val -> convertLiteral(((Literal<?>) val)))
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()));
+          } else {
+            return null;
+          }
+
+        case NOT:
+          Not notPredicate = (Not) predicate;
+          Predicate childPredicate = notPredicate.child();
+          if (childPredicate.name().equals(IN)) {
```

Comment:
Shall we add also `isSupportedInPredicate(childPredicate)` to the condition?

Reply:
Added. Thanks!